components/identifier-generator/back/tests/EndToEnd/Infrastructure/Controller/GetFamiliesControllerEndToEnd.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Test\Pim\Automation\IdentifierGenerator\EndToEnd\Infrastructure\Controller;
+
+use Akeneo\Test\Integration\Configuration;
+use Akeneo\Test\Pim\Automation\IdentifierGenerator\EndToEnd\ControllerEndToEndTestCase;
+use PHPUnit\Framework\Assert;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
+ * @license   https://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class GetFamiliesControllerEndToEnd extends ControllerEndToEndTestCase
+{
+    /** @test */
+    public function it_should_redirect_on_non_xhr_request(): void
+    {
+        $this->loginAs('Julia');
+        $this->callRoute('akeneo_identifier_generator_get_families', [
+            'HTTP_X-Requested-With' => 'toto',
+        ]);
+        $response = $this->client->getResponse();
+        Assert::AssertSame(Response::HTTP_FOUND, $response->getStatusCode());
+        Assert::assertTrue($response->isRedirect('/'));
+    }
+
+    /** @test */
+    public function it_return_http_forbidden_without_acl(): void
+    {
+        $this->removeAclFromRole('action:pim_enrich_family_index');
+        $this->loginAs('admin');
+
+        $this->callGetRouteWithQueryParam('akeneo_identifier_generator_get_families', ['codes' => 'familyA1,familyA2']);
+        $response = $this->client->getResponse();
+
+        Assert::AssertSame(Response::HTTP_FORBIDDEN, $response->getStatusCode());
+    }
+
+    /** @test */
+    public function it_returns_a_list_of_families_filtered_by_codes(): void
+    {
+        $this->loginAs('Julia');
+
+        $this->callGetRouteWithQueryParam('akeneo_identifier_generator_get_families', ['codes' => 'familyA1,familyA2']);
+        $response = $this->client->getResponse();
+
+        $expected = [
+            [
+            'code' => 'familyA1',
+            'labels' => [
```

Comment:
```suggestion
                'code' => 'familyA1',
                'labels' => [
```
My eyes are burning :trollface: 
Don't forget to run linter